restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/NumberSelectorService.java

```diff
@@ -279,7 +279,7 @@ public NumberSelectionResult searchNumberWithResult(String phone,
         if (numberfound.number == null) {
             //only use regex if perfect match didnt worked
             if (destinationOrganizationSid != null
-                    && destinationOrganizationSid.equals(sourceOrganizationSid)
+                    &&  (sourceOrganizationSid == null || destinationOrganizationSid.equals(sourceOrganizationSid))
```

Comment:
not sure if we should provide debug messages on why we are not evaluating regexes (if some condition is not met), considering all issues lately...

Reply:
I can add some debug log statements